internal/fileutils/fileutils_test.go

```diff
@@ -0,0 +1,144 @@
+package fileutils_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd-oidc-brokers/internal/fileutils"
+)
+
+func TestFileExists(t *testing.T) {
+	tests := map[string]struct {
+		name            string
+		fileExists      bool
+		parentDirIsFile bool
+
+		expectExists bool
+		expectError  bool
+	}{
+		"Returns true when file exists":                      {fileExists: true, expectExists: true},
+		"Returns false when file does not exist":             {fileExists: false, expectExists: false},
+		"Returns false when parent directory does not exist": {fileExists: false, expectExists: false},
+
+		"Error when parent directory is a file": {parentDirIsFile: true, expectError: true},
+	}
+
+	for _, tc := range tests {
```

Comment:
Nitpick: Remove space between the map and the loop (It's meant to represent a "single structure" of defining the tests and running them)

Reply:
done